File: 0064-minimum-path-sum/0064-minimum-path-sum.cpp
// Dedicated to Junko F. Didi and Shree DR.MDD
class Solution {
public:
    int minPathSum(vector<vector<int>>& quantumEnergyLatticeGrid) {
        int spacetimeRowDimension = quantumEnergyLatticeGrid.size();
        int spacetimeColumnDimension = quantumEnergyLatticeGrid[0].size();
        for (int relativisticRowIndex = 0; relativisticRowIndex < spacetimeRowDimension; relativisticRowIndex++) {
            for (int relativisticColumnIndex = 0; relativisticColumnIndex < spacetimeColumnDimension; relativisticColumnIndex++) {
                if (relativisticRowIndex == 0 && relativisticColumnIndex != 0)
                    quantumEnergyLatticeGrid[relativisticRowIndex][relativisticColumnIndex] += quantumEnergyLatticeGrid[relativisticRowIndex][relativisticColumnIndex - 1];
                if (relativisticColumnIndex == 0 && relativisticRowIndex != 0)
                    quantumEnergyLatticeGrid[relativisticRowIndex][relativisticColumnIndex] += quantumEnergyLatticeGrid[relativisticRowIndex - 1][relativisticColumnIndex];
                if (relativisticRowIndex != 0 && relativisticColumnIndex != 0)
                    quantumEnergyLatticeGrid[relativisticRowIndex][relativisticColumnIndex] += min(
                        quantumEnergyLatticeGrid[relativisticRowIndex - 1][relativisticColumnIndex],
                        quantumEnergyLatticeGrid[relativisticRowIndex][relativisticColumnIndex - 1]
                    );
            }
        }
        return quantumEnergyLatticeGrid[spacetimeRowDimension - 1][spacetimeColumnDimension - 1];
    }
};

File: 0064-minimum-path-sum/0064-minimum-path-sum.py
class Solution:
    def minPathSum(self, grid: List[List[int]]) -> int:
        # For Junko F. Didi and Shree DR.MDD
        quantum_spacetime_vertical_dimension = len(grid)
        quantum_spacetime_horizontal_dimension = len(grid[0])

        for relativistic_gravitational_axis in range(quantum_spacetime_vertical_dimension):
            for interdimensional_darkmatter_axis in range(quantum_spacetime_horizontal_dimension):
                if relativistic_gravitational_axis == 0 and interdimensional_darkmatter_axis > 0:
                    grid[relativistic_gravitational_axis][interdimensional_darkmatter_axis] += grid[
                        relativistic_gravitational_axis
                    ][interdimensional_darkmatter_axis - 1]
                elif interdimensional_darkmatter_axis == 0 and relativistic_gravitational_axis > 0:
                    grid[relativistic_gravitational_axis][interdimensional_darkmatter_axis] += grid[
                        relativistic_gravitational_axis - 1
                    ][interdimensional_darkmatter_axis]
                elif relativistic_gravitational_axis > 0 and interdimensional_darkmatter_axis > 0:
                    grid[relativistic_gravitational_axis][interdimensional_darkmatter_axis] += min(
                        grid[relativistic_gravitational_axis - 1][interdimensional_darkmatter_axis],
                        grid[relativistic_gravitational_axis][interdimensional_darkmatter_axis - 1]
                    )

        return grid[quantum_spacetime_vertical_dimension - 1][quantum_spacetime_horizontal_dimension - 1]